client/extensions/zoninator/components/settings/zone-creator/index.jsx

```diff
@@ -4,66 +4,50 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { reduxForm } from 'redux-form';
 import { localize } from 'i18n-calypso';
 import { flowRight } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import CompactCard from 'components/card';
-import FormFieldset from 'components/forms/form-fieldset';
-import FormButton from 'components/forms/form-button';
-import FormLabel from 'components/forms/form-label';
 import HeaderCake from 'components/header-cake';
-import ReduxFormTextarea from 'components/redux-forms/redux-form-textarea';
-import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
-import SectionHeader from 'components/section-header';
-import { getSelectedSiteSlug } from 'state/ui/selectors';
-import { settingsPath } from '../../../app/util';
-
-const form = 'extensions.zoninator.newZone';
-
-const ZoneCreator = ( { siteSlug, translate } ) => (
-	<div>
-		<HeaderCake backHref={ `${ settingsPath }/${ siteSlug }` }>
-			{ translate( 'Add a zone' ) }
-		</HeaderCake>
-
-		<form>
-			<SectionHeader label={ translate( 'New zone' ) }>
-				<FormButton compact>{ translate( 'Save' ) }</FormButton>
-			</SectionHeader>
-			<CompactCard>
-				<FormFieldset>
-					<FormLabel htmlFor="zoneName">{ translate( 'Zone name' ) }</FormLabel>
-					<ReduxFormTextInput name="zoneName" />
-				</FormFieldset>
-
-				<FormFieldset>
-					<FormLabel htmlFor="zoneDescription">{ translate( 'Zone description' ) }</FormLabel>
-					<ReduxFormTextarea name="zoneDescription" />
-				</FormFieldset>
-			</CompactCard>
-		</form>
-	</div>
-);
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import ZoneDetailsForm from '../../forms/zone-details-form';
+import { addZone } from '../../../state/zones/actions';
+
+const ZoneCreator = ( {
+	saveZone,
+	siteId,
+	siteSlug,
+	translate,
+} ) => {
+	const save = ( form, data ) => saveZone( siteId, form, data );
```

Comment:
Shouldn't this method be part of `<ZoneDetailsForm />`?

Reply:
Oh, there actually _is_ a `save` method in `ZoneDetailsForm`. Now I'm a bit confused... 😄

Reply:
The `save` method in `ZoneDetailsForm` triggers the `onSubmit` callback passed from `ZoneCreators` and adds a `form` argument.

Reply:
Alternatively, you could pass `saveZone` and `siteId` as props to `ZoneDetailsForm` and just call it directly from there instead of going through the parent component. I believe I did things this way in WPJM because I was dealing with more than one form. As a bonus, by eliminating this function we could keep `ZoneCreator` as a functional component and my next comment could be ignored.